src/SimpleNonlinearSolve.jl

```diff
@@ -81,7 +82,18 @@ function SciMLBase.solve(prob::NonlinearProblem, alg::AbstractSimpleNonlinearSol
         p === nothing, alg, args...; prob.kwargs..., kwargs...)
 end
 
-function __internal_solve_up(_prob::NonlinearProblem, sensealg, u0, u0_changed,
+function SciMLBase.solve(prob::Union{ImmutableNonlinearProblem}, alg::AbstractSimpleNonlinearSolveAlgorithm,
+    args...; sensealg = nothing, u0 = nothing, p = nothing, kwargs...)
```

Comment:
The conversion should go here.

Reply:
I moved the conversion to inside the solve in the last commit. Now I'm getting a method error when precompiling due to the startup workload. It is complaining about this line when solving a problem that has been converted and its not clear to me how to fix it: https://github.com/SciML/SimpleNonlinearSolve.jl/blob/2e609d9fc952881f124e21bb8181a3a5c4e3b91a/src/nlsolve/lbroyden.jl#L185